.github/workflows/docker-build-push-production.yml

```
@@ -0,0 +1,90 @@
+name: Docker build and push
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  AWS_REGION: ca-central-1
+  FUNCTION_PREFIX: gcds-api 
+  GITHUB_SHA: ${{ github.sha }}
+  REGISTRY: 307395567143.dkr.ecr.ca-central-1.amazonaws.com/gc-design-system/api 
+
+permissions:
+  id-token: write
+  contents: write
+  pull-requests: write
+  actions: write
+  checks: write
+  statuses: write
+  security-events: write
+
+jobs:
+  changes:
+    runs-on: ubuntu-latest
+    outputs:
+      images: ${{ steps.filter.outputs.changes }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - uses: dorny/paths-filter@4512585405083f25c027a35db413c2b3b9006d50 # tag=v2.11.1
+        id: filter
+        with:
+          filters: |
+            api: 'api/**'
+
+  build-push:
+    if: needs.changes.outputs.images != '[]'
+    runs-on: ubuntu-latest
+    needs: changes
+    strategy:
+      fail-fast: false
+      matrix:
+        image: ${{ fromJSON(needs.changes.outputs.images) }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Configure AWS credentials using OIDC
+        uses: aws-actions/configure-aws-credentials@010d0da01d0b5a38af31e9c3470dbfdabdecca3a # v4.0.1
+        with:
+          role-to-assume: arn:aws:iam::307395567143:role/gcds-terraform-apply
```

My mistake for this - I forgot these workflows were in the `gcds-docs` repo so I'll need to create a new OIDC role for you that will be called:
```suggestion
          role-to-assume: arn:aws:iam::307395567143:role/gcds-docs-apply
```